### DISCUSSION TITLE
rose stem tests failing

### DISCUSSION THREAD

**Discussion by cehalliwell** [MetOffice/simulation-systems#288] [LFRic]
I created an LFRic branch at vn1.1 and, before implementing any code changes, I followed instructions in

[GhaspSupport/PhysicsHandsOn – LFRic](https://code.metoffice.gov.uk/trac/lfric/wiki/GhaspSupport/PhysicsHandsOn)

and ran the developer group rose stem tests via:

rose stem --group=developer

Many of the tasks in BUILD failed as well as several others.  

I repeated the branch creation and check out (just in case!) and, once again, tried to run the above rose stem test.  This time, I got the following error message:

WorkflowFilesError: Nested run directories not allowed - cannot install workflow in '/home/h03/apcr/cylc-run/vn1.1_1DTKE_LFric/run1' as '/home/h03/apcr/cylc-run/vn1.1_1DTKE_LFric' is already a valid run directory.

I feel like I might be missing an important step(s).  I haven't made any code changes yet. Can anyone advise?  Thanks

First branch is: https://code.metoffice.gov.uk/svn/lfric_apps/main/branches/dev/carolhalliwell/vn1.1_1DTKE_scheme
Second branch is: https://code.metoffice.gov.uk/svn/lfric_apps/main/branches/dev/carolhalliwell/vn1.1_1DTKE_LFric

**Comment by jennyhickson** [comment]
Hi,

I'm not sure how up to date the instructions in that physics hands on page are. There are full development practices available here now, and this specifically is for running the LFRic Apps test suite https://metoffice.github.io/simulation-systems/WorkingPractices/TestSuites/lfric_apps.html

Most notably, the LFRic Apps rose-stem requires use of cylc 8 and the error you're seeing would suggest to me that you've run it once with cylc7 and once with cylc8 which generally get things confused. I'd clean up cylc-run/vn1.1_1DTKE_LFRic and have another go using cylc8 and see if that sorts it for you.

**Reply by jennyhickson** [reply]
I've just taken a look and the Physics Hands On page has had some recent updates, so should be good. Hopefully getting the right cylc version is all you need.

**Reply by cehalliwell** [reply]
Hi @jennyhickson.  I think the error message I posted was different to my initial failures - probably because I was using the wrong cylc version at some point.

 Just to be on the safe side, I opened a new terminal, created a new branch, ran export CYLC_VERSION=8 and tried running the rose stem developer group tests.  The build tasks are still failing though :-(

**Comment by cehalliwell** [comment]
These are the exact commands I used after opening a new terminal.  The cylc version is shown to be 8.3.6  The same tasks fail every time I do this (mainly in BUILD).  Am I missing a step to set something up before I run these commands?

fcm bc --ticket=338 1DTKE_scheme_in_LFRic fcm:lfric_apps.x_tr@vn1.1
fcm co https://code.metoffice.gov.uk/svn/lfric_apps/main/branches/dev/carolhalliwell/vn1.1_1DTKE_scheme_in_LFRic
export CYLC_VERSION=8
cylc --version 
rose stem --group=developer
cylc play vn1.1_1DTKE_scheme_in_LFRic
cylc gui

**Reply by jennyhickson** [reply]
Hi Carol,

Those commands look fine to me, there shouldn't be anything else to do. Please could you give an example of the errors you're seeing? Thanks

**Reply by cehalliwell** [reply]
Hi Jenny.

It's difficult to know where to start.  All the output files are in ~/cylc-run/vn1.1_1DTKE_scheme_in_LFRic/run2/log/job/1/
and most of the tasks starting with "build" fail.  I've picked a couple of random tasks:

task build_gungho_model_spice_intel_fast-debug-32bit
Messages and errors include:

Unsupported API 'lfric' specified. Supported APIs are ['dynamo0.3', 'gocean1.0', 'nemo'].
make[2]: *** [/var/tmp/apcr-3206510/intel_fast-debug-32bit/vn1.1_1DTKE_scheme_in_LFRic/run2/gungho_model/algorithm/sci_geometric_constants_mod.f90] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [/var/tmp/apcr-3206510/intel_fast-debug-32bit/vn1.1_1DTKE_scheme_in_LFRic/run2/gungho_model/algorithm/sci_checksum_alg_mod.f90] Error 1
make[2]: *** [/var/tmp/apcr-3206510/intel_fast-debug-32bit/vn1.1_1DTKE_scheme_in_LFRic/run2/gungho_model/algorithm/sci_fem_constants_mod.f90] Error 1
make[2]: *** [/var/tmp/apcr-3206510/intel_fast-debug-32bit/vn1.1_1DTKE_scheme_in_LFRic/run2/gungho_model/algorithm/sci_assign_field_random_range_alg_mod.f90] Error 1
make[1]: *** [import-science] Error 2

while task fcm_make2_lfricinputs_lfric2um_spice_gfortran-gcc_safe
show something like:

[FAIL] operator_real32_mod.mod: bad or missing dependency (type=1.include)
[FAIL]     required by: id_r32_operator_pair_mod.o
[FAIL]     required by: id_r32_operator_pair_mod.mod
[FAIL]     required by: inventory_by_mesh_mod.o
[FAIL]     required by: inventory_by_mesh_mod.mod
[FAIL]     required by: lfricinp_runtime_constants_mod.o
[FAIL]     required by: lfricinp_runtime_constants_mod.mod
[FAIL]     required by: lfricinp_lfric_driver_mod.o
[FAIL]     required by: lfricinp_lfric_driver_mod.mod
[FAIL]     required by: lfric2um.o
[FAIL]     required by: lfric2um.exe

**Reply by cehalliwell** [reply]
sorry, full path of output files is /home/h03/apcr/cylc-run/vn1.1_1DTKE_scheme_in_LFRic/run2/log/job/1/

**Reply by andrewcoughtrie** [reply]
Hi Carol,

I think your branch might be picking up the wrong version of lfric_core, can you have a look at the version of lfric_core in the `dependencies.sh` file.

Andy

**Reply by cehalliwell** [reply]
Hi Andy,
The relevant command in dependencies.sh is 
export lfric_core_rev=core1.1

**Reply by james-bruten-mo** [reply]
Hi Carol, can you make sure you can ssh to your own vdi passwordless. Looking at the vdi assignments that'd be `ssh vld417`. Once you can do that with no interaction you should be ok to go

**Reply by cehalliwell** [reply]
@james-bruten-mo ssh from where?  Thanks
ssh vld417 works already without a password from VDI terminal and HPCs.

**Reply by james-bruten-mo** [reply]
Just from your vdi command line

**Reply by cehalliwell** [reply]
Yes, this command already works without a password from vdi command line.

**Reply by cehalliwell** [reply]
> Hi Andy, The relevant command in dependencies.sh is export lfric_core_rev=core1.1

@andrewcoughtrie what version should I be picking up?  Is this correct?

**Reply by andrewcoughtrie** [reply]
It should pick up that one, I think the problem is more related to what @james-bruten-mo mentioned.

**Comment by cehalliwell** [comment]
Thanks to everyone for their suggestions/help.  Although I didn't solve this problem when using LFRic version 1.1, I am able to run rose stem tests at vn2.0 so will be using this version. 
Carol
